File: LeetCode/Pascal's Triangle.py
class Solution:
    def generate(self, n: int) -> List[List[int]]:
        if (n ==1): res = [[1]]
        elif (n==2): res= [[1],[1,1]]
        else:
            res = [[1],[1,1]]
            for i in range(3, n+1):
                t = [1]
                for j in range(0,len(res[i-2])-1):
                    t.append(int(res[i-2][j] + res[i-2][j+1]))
                t.append(1)
                res.append(t)
        return res

File: LeetCode/Arrays/Pascal's Triangle.java
import java.util.*;
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> t = new ArrayList<>();
        t.add(1);
        res.add(t);
        numRows -=1;
        int p = 0;
        while (numRows>0){
            List<Integer> l = new ArrayList<>();
            List<Integer> prev = res.get(p);
            if (res.get(p).size()==0){
                l.add(1);
                l.add(1);
                res.add(l);
                p++;
            }else{
                l.add(1);
                for (int i=0; i<prev.size()-1; i++){
                    l.add(prev.get(i)+prev.get(i+1));
                }
                l.add(1);
                p++;
            }
            res.add(l);
            numRows--;
        }
        return res;
    }
}
